pkgs/applications/networking/instant-messengers/element/element-desktop.nix

```
@@ -40,7 +41,7 @@ mkYarnPackage rec {
   nativeBuildInputs = [ makeWrapper ];
 
   seshat = callPackage ./seshat { inherit CoreServices; };
-  keytar = callPackage ./keytar { inherit Security AppKit; };
+  keytar = if useKeytar then (callPackage ./keytar { inherit Security AppKit; }) else null;
```

I'm unsure if this is the appropriate method to do this.

---

Nix is lazy so we don't need to do this. It is a very common miss conception that optional inputs somehow can or must be null which is false.

```suggestion
  keytar = callPackage ./keytar { inherit Security AppKit; };
```

---

Huh, but `keytar` is turned into `$keytar` inside of the derivation scripts, how can nix know if it is used or not?

---

If line 55 evals to false the string is never inserted. Also instead of $keytar ${keytar} should be used to insert the path at nix eval time.

---

If line 55 evals to false the string is never inserted. Also instead of $keytar ${keytar} should be used to insert the path at nix eval time.